File: sdk/python/examples/streaming_chat.py
"""
Streaming Chat Example

Run with: python examples/streaming_chat.py
"""

import os
import sys
from infinity_assistant import InfinityAssistantClient

def main():
    client = InfinityAssistantClient(
        api_key=os.getenv("INFINITY_ASSISTANT_API_KEY", "ia_your_api_key_here")
    )

    try:
        print("Starting streaming chat...\n")

        full_response = ""

        for chunk in client.chat_stream(
            message="Write a short story about a robot learning to paint",
            mode="assist"
        ):
            if chunk["type"] == "text":
                content = chunk.get("content", "")
                sys.stdout.write(content)
                sys.stdout.flush()
                full_response += content
            elif chunk["type"] == "metadata":
                print("\n\n[Metadata]", chunk.get("metadata"))
            elif chunk["type"] == "done":
                print("\n\n[Stream complete]")
                break
            elif chunk["type"] == "error":
                print("\n\n[Error]", chunk.get("error"))
                break

        print("\n\nFull response length:", len(full_response))

    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: sdk/typescript/examples/streaming-chat.ts
/**
 * Streaming Chat Example
 * 
 * Run with: npx tsx examples/streaming-chat.ts
 */

import { InfinityAssistantClient } from '../src';

async function main() {
  const client = new InfinityAssistantClient({
    apiKey: process.env.INFINITY_ASSISTANT_API_KEY || 'ia_your_api_key_here',
  });

  try {
    console.log('Starting streaming chat...\n');

    let fullResponse = '';

    for await (const chunk of client.chatStream({
      message: 'Write a short story about a robot learning to paint',
      mode: 'assist',
    })) {
      if (chunk.type === 'text') {
        process.stdout.write(chunk.content || '');
        fullResponse += chunk.content || '';
      } else if (chunk.type === 'metadata') {
        console.log('\n\n[Metadata]', chunk.metadata);
      } else if (chunk.type === 'done') {
        console.log('\n\n[Stream complete]');
        break;
      } else if (chunk.type === 'error') {
        console.error('\n\n[Error]', chunk.error);
        break;
      }
    }

    console.log('\n\nFull response length:', fullResponse.length);
  } catch (error: any) {
    console.error('Error:', error.message);
  }
}

main();

